### PR TITLE
refactor(debug): switch test from TCB to use TestBed

### PR DESCRIPTION
**Pull request by cnishina**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format

**What kind of change does this PR introduce?**
Refactoring (no functional changes, no api changes)

**What is the current behavior?** (You can also link to an open issue here)
debug_node_spec uses TestComponentBuilder

**What is the new behavior?**
Use TestBed

**Does this PR introduce a breaking change?** (check one with "x")
No

### CODE REVIEW

**Comment by IgorMinar** [comment]
@cnishina can you please rebase?

**Comment by cnishina** [comment]
@IgorMinar done.

File: modules/@angular/core/test/debug/debug_node_spec.ts [hunk 1]
```diff
@@ -176,255 +168,195 @@ class TestApp {
...
+          MyDir,
+          ParentComp,
+          TestApp,
+          UsingFor,
+        ],
+        providers: [
+          Logger,
+        ]
+      });
+      TestBed.compileComponents();
```
Comment (juliemr) [inline modules/@angular/core/test/debug/debug_node_spec.ts]: Remove `compileComponents` or put inside an async block.

**Comment by juliemr** [comment]
Same comment as other PR, after that, ready to merge.

**Comment by IgorMinar** [comment]
please squash

**Comment by cnishina** [comment]
Done.

**Pull request merged**